gdsfactory/functions.py

```diff
@@ -262,58 +260,43 @@ def add_settings_label(
 
 
 def add_marker_layer(
```

Comment:
if this is intended to be used a as a decorator it should apply to the component in place

Reply:
Good catch, I'll update